main/prelims.tex

```diff
@@ -11,3 +11,25 @@ \section{Some section}
 The \emph{von Neumann model} of a computer, also known as the \emph{Princeton architecture} is an architecture for digital computers, which consists of a processing units, containing an ALU and processing registers; a control unit consisting of an instruction register and a program counter; a memory unit which stores both data and instructions; and input-and-output mechanisms.
 \end{definition}
 
+\section{Acronyms and abbreviations}
+
+\newif\ifabbreviations
+\ifbool{iitthesis-abbrevs}{
+  \abbreviationstrue
+}{
+  \abbreviationsfalse
+}
+
+\ifabbreviations
+  When defining a \gls{tla} for the first time (or an acronym with any number of
+  letters), the \verb|\gls{tla}| macro of the \texttt{glossaries-extra} package
+  generates its full form, with its acronym in parentheses. The acronym is defined
+  in the \texttt{abbrevs.tex} file. The next time a \gls{tla} is used, only its
+  short form is printed. \Glspl{etla} can be capitlized or use the plural form
+  with macros such as \verb|\Gls| or \verb|\glspl|.
+
+  In addition, the \verb|\gls| macro adds the pages where a term is used to the
+  \nameref{chap:abbrev} \autoref{chap:abbrev}.
+\else
+  The Notation and Abbreviations Chapter is disabled.
```

Comment:
I understand the motivation to have this conditional code. But - this is a sample thesis. Let's assume a relatively naive author and just take the `\ifabbreviations` path here and drop the the `\else` case.

Reply:
The problem is that the \gls macro is not defined when `noabbrevs` is used.

Reply:
It's not a problem. It's ok that the `\gls`'s will trigger errors if `noabbrevs` is added. In fact, I would mention it in the text, i.e. say that in writing this paragraph we are using the `\gls` command, which is not necessarily available if `noabbrevs` is specified.